common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ShuffleTransportContext.java

```diff
@@ -52,7 +53,8 @@
  * */
 public class ShuffleTransportContext extends TransportContext {
   private static final Logger logger = LoggerFactory.getLogger(ShuffleTransportContext.class);
-  private static final ShuffleMessageDecoder SHUFFLE_DECODER =
+  @VisibleForTesting
+  protected static ShuffleMessageDecoder SHUFFLE_DECODER =
       new ShuffleMessageDecoder(MessageDecoder.INSTANCE);
```

Comment:
Instead of exposing the variable, add a method to reinitialize it - and annotate as for use by tests.